lib/puppet/indirector/catalog/rest.rb

```diff
@@ -3,4 +3,48 @@
 
 class Puppet::Resource::Catalog::Rest < Puppet::Indirector::REST
   desc "Find resource catalogs over HTTP via REST."
+
+  def find(request)
+    return super unless use_http_client?
+
+    # URL encoded facts and facts_format are passed as indirector
+    # request options, so we have to reverse that (unescape, then parse),
+    # and pass a facts object to the http client.
+    format = request.options[:facts_format]
+    if format
+      formatter = Puppet::Network::FormatHandler.format_for(format)
+      facts = formatter.intern(Puppet::Node::Facts, CGI.unescape(request.options[:facts]))
```

Comment:
This reverses what is done https://github.com/puppetlabs/puppet/blob/da4bea2d260a577b3415fac83a881d5c1c1bcdb1/lib/puppet/configurer/fact_handler.rb#L44-L48, including converting `%20` back into spaces.

Reply:
This code is no longer needed, removed.